lib/routes/totp.js

```diff
@@ -53,41 +53,37 @@ module.exports = (log, db, mailer, customs, config) => {
           })
         }
       },
-      handler(request, reply) {
+      handler: async function (request, h) {
         log.begin('totp.create', request)
 
-        let response
         const sessionToken = request.auth.credentials
         const uid = sessionToken.uid
-
-        customs.check(request, sessionToken.email, 'totpCreate')
-          .then(createTotpToken)
-          .then(emitMetrics)
-          .then(createResponse)
-          .then(() => reply(response), reply)
-
         const authenticator = new otplib.authenticator.Authenticator()
         authenticator.options = otplib.authenticator.options
         const secret = authenticator.generateSecret()
+        await customs.check(request, sessionToken.email, 'totpCreate')
+        await createTotpToken()
+        await emitMetrics()
+        const response = await createResponse()
+        return response
+
 
         function createTotpToken() {
           if (sessionToken.tokenVerificationId) {
             throw errors.unverifiedSession()
           }
-
           return db.createTotpToken(uid, secret, 0)
         }
 
-        function createResponse() {
+        async function createResponse() {
```

Comment:
@deeptibaghel this is the only `createResponse` that is async, the rest in other routes just return an object. @vbudhram thoughts about that? Does that lead to inconsistency or should we just leave it?

Reply:
@deeptibaghel @vladikoff If it isn't too hard, I think we should make it consistent with other routes.